crates/cubecl-runtime/src/memory_management/memory_lock.rs

```diff
@@ -1,20 +1,34 @@
-use alloc::collections::BTreeSet;
-
 use crate::storage::StorageId;
+use alloc::collections::BTreeSet;
 
 /// A set of storage buffers that are 'locked' and cannot be
 /// used for allocations currently.
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct MemoryLock {
     locked: BTreeSet<StorageId>,
+    flush_threshold: usize,
 }
 
 impl MemoryLock {
+    /// Create a new memory lock with the given flushing threshold.
+    pub fn new(flush_threshold: usize) -> Self {
+        Self {
+            locked: Default::default(),
+            flush_threshold,
+        }
+    }
     /// Check whether a particular storage ID is locked currently.
     pub fn is_locked(&self, storage: &StorageId) -> bool {
         self.locked.contains(storage)
     }
 
+    /// Whether the flushing threshold has been reached.
+    pub fn should_flush(&self) -> bool {
+        // For now we only consider the number of handles locked, but we may consider the amount in
+        // bytes at some point.
+        self.locked.len() >= self.flush_threshold
+    }
```

Comment:
Regarding its usage, I suggest renaming to `has_reach_threshold`. At least, when use in `WgpuStream`, we don't flush the memory lock. The current name suggests that after this function returns `true`, we should flush the memory lock.